ably/sharded_task.go

```diff
@@ -0,0 +1,159 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/rand"
+	"strconv"
+	"time"
+
+	"github.com/ably-forks/boomer"
+	"github.com/ably/ably-go/ably"
+)
+
+func generateChannelName(testConfig TestConfig, number int) string {
+	return "sharded-test-channel-" + strconv.Itoa(number%testConfig.NumChannels)
+}
+
+func randomChannelName(testConfig TestConfig) string {
```

Comment:
It would be more deterministic if we had some sort of `sharded worker number % NumChannels` here. The random distribution of subscribers could add a lot of variability to the reported statistics.